src/printer.js

```diff
@@ -2903,6 +2909,25 @@ function isFlowNodeStartingWithDeclare(node, options) {
     .match(/declare\s*$/);
 }
 
+function hasDanglingComments(node) {
```

Comment:
Yeah, you're totally right and anyway I'm going to refactor it to only use dangling comments when there are no params.